lib/evaluate/fold.cc

```diff
@@ -526,6 +526,26 @@ Expr<Type<TypeCategory::Integer, KIND>> UBOUND(FoldingContext &context,
   return Expr<T>{std::move(funcRef)};
 }
 
+template<typename T>
+Expr<T> FoldMINorMAX(
+    FoldingContext &context, FunctionRef<T> &&funcRef, Ordering order) {
+  std::vector<Constant<T> *> constantArgs;
+  for (auto &arg : funcRef.arguments()) {
+    if (auto *cst{FoldConvertedArg<T>(context, arg)}) {
+      constantArgs.push_back(cst);
+    } else {
+      return Expr<T>(std::move(funcRef));
+    }
+  }
+  CHECK(constantArgs.size() >= 2);
```

Comment:
What if a program contains `MIN(x)`?  Are you sure that we won't try to fold it later, or accept it as an extension?

Reply:
You are right, I do not need to enforce `>=2` when `>=1` is sufficient for what is done here. Changed.